### PR TITLE
gather_nd function docstrings done.

### PR DESCRIPTION
**Pull request by zaeemansari70**

### CODE REVIEW

**Comment by zaeemansari70** [comment]
#1637

File: ivy/functional/ivy/general.py [hunk 1]
```diff
@@ -2308,8 +2308,79 @@ def gather_nd(
...
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), \
+                          b=ivy.array([4., 5., 6.]))
+    >>> y = ivy.Container(a=ivy.array([0]), \
+                          b=ivy.array([2]))
+    >>> print(ivy.gather_nd(x, y))
+    {
+     	a: ivy.array(0.),
```
Comment (VedPatwardhan) [inline ivy/functional/ivy/general.py:2379]: This line causes a lint failure as it seems to have a mix of spaces and tabs, please update that to have only spaces.

File: ivy/array/general.py [hunk 2]
```diff
@@ -39,3 +39,24 @@ def all_equal(
...
+
+        Examples
+        --------
+        >>> x = ivy.array([1, 2, 3])
+        >>> y = ivy.array([4, 5, 6])
+        >>> z = x.gather_nd(y)
+        >>> print(z)
+        ivy.array(3)
+        """
+        return ivy.gather_nd(self._data, indices, out=out)
```
Comment (VedPatwardhan) [inline ivy/array/general.py:62]: Please update this line to replace `self._data` with `self`, as `self._data` refers to native array while we want to call the function with the ivy array.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @zaeemansari70, please make the changes where I've mentioned. Also please update the docstring so that it's only indented with spaces, because otherwise it causes the numpy style docstring test to fail. 
Let me know if you have any questions. Thanks 🙂

File: ivy/array/general.py [hunk 3]
```diff
@@ -39,3 +39,24 @@ def all_equal(
...
+    ) -> Union[ivy.Array, ivy.NativeArray]:
+        """
+        ivy.Array instance method variant of ivy.gather_nd. This method simply wraps the
+        function, and so the docstring for ivy.gather_nd also applies to this method
+        with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.array([1, 2, 3])
+        >>> y = ivy.array([4, 5, 6])
```
Comment (VedPatwardhan) [inline ivy/array/general.py:57]: Please update `y` which is being passed to the function as index cannot be larger than 2 for the current value of `x`

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @zaeemansari70, the docstring tests are failing because of passing a `device` argument to the function in some examples. The function doesn't accept a `device` argument currently, so please remove it from all examples which pass a `device` argument to the function. Thanks 🙂

File: ivy/container/general.py [hunk 4]
```diff
@@ -177,3 +177,97 @@ def all_equal(
...
+    ) -> ivy.Container:
+        """
+        ivy.Container instance method variant of ivy.gather_nd.
+        This method simply wraps the function, and so the docstring
+        for ivy.gather_nd also applies to this method
+        with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([1, 2, 3]),\
```
Comment (VedPatwardhan) [inline ivy/container/general.py:254]: Hi @zaeemansari70, please update this example as the output of the example is incorrect.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/general.py [hunk 5]
```diff
@@ -2309,8 +2309,83 @@ def gather_nd(
...
+    Examples
+    --------
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([0., 1., 2., 3., 4., 5., 6.])
+    >>> y = ivy.array([1])
+    >>> print(ivy.gather_nd(x, y))
+    ivy.array(1.)
+
+    >>> x = ivy.array([[0., 1., 2.], \
```
Comment (VedPatwardhan) [inline ivy/functional/ivy/general.py:2321]: Based on the logs of the test, the output of this example seems to be incorrect as well, please execute it on your machine and then update the result accordingly.

**Pull request merged**